util/webhook/webhook.go

```diff
@@ -258,6 +249,27 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}) {
 	}
 }
 
+// getWebUrlRegex compiles a regex that will match any targetRevision referring to the same repo as the given webURL.
+// webURL is expected to be a URL from an SCM webhook payload pointing to the web page for the repo.
+func getWebUrlRegex(webURL string) *regexp.Regexp {
+	urlObj, err := url.Parse(webURL)
+	if err != nil {
+		log.Warnf("Failed to parse repoURL '%s'", webURL)
+		return nil
+	}
+
+	regexEscapedHostname := regexp.QuoteMeta(urlObj.Hostname())
+	regexEscapedPath := regexp.QuoteMeta(urlObj.Path[1:])
+	regexpStr := fmt.Sprintf(`(?i)^(http://|https://|\w+@|ssh://(\w+@)?)%s(:[0-9]+|)[:/]%s(\.git)?$`, regexEscapedHostname, regexEscapedPath)
```

Comment:
The only change to the pattern (besides escaping the inputs) is to add `^` and `$`.